app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,56 @@
+require 'report_email_sender'
+
+class ReportsController < ApplicationController
+  def new
+    @report = Report.new
+  end
+
+  def create
+    keywords = params['keywords'].values.select { |k| !k.empty? }
+    if keywords.empty?
```

Comment:
It might also mean we don't need all of the manual `@error` wrangling in the controllers/views